test/functional/ui/sign_spec.lua

```diff
@@ -607,4 +607,63 @@ describe('Signs', function()
     eq(6, infos[1].textoff)
     eq(6, infos[2].textoff)
   end)
+
+  it('auto width updated in all windows after sign placed in on_win #31438', function()
+    exec_lua([[
+      vim.cmd.call('setline(1, range(1, 500))')
+      vim.cmd('wincmd s | wincmd v | wincmd j | wincmd v')
+
+      local ns_id, mark_id = vim.api.nvim_create_namespace('test'), nil
+
+      -- Add decoration which possibly clears all extmarks and adds one on line 499
+      local on_win = function(_, winid, bufnr, toprow, botrow)
+        if _G.needs_clear then
+          vim.api.nvim_buf_clear_namespace(bufnr, ns_id, 0, -1)
+          _G.needs_clear = false
```

Comment:
Should this also test the `:lua needs_clear = true` case? As it seems to have different behavior before this PR (all windows without signcolumn vs windows before target are without signcolumn).